src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,296 @@
+        final String DELIMITER = "--";
+        int doneValue;
+        FileWriter beginWrite = new FileWriter("data/duke.txt");
+
+        beginWrite.write("");
+        beginWrite.close();
+
+        FileWriter fw = new FileWriter("data/duke.txt", true);
+
+        for (int i = 0; i < userInputsCount; i++) {
+            if (userInputs.get(i).getCode().equals("T")) {
+                String lineToAppend = userInputs.get(i).getCode() + DELIMITER + userInputs.get(i).getDoneValue() +
+                        DELIMITER + userInputs.get(i).getDescription();
+                fw.write(lineToAppend);
+                fw.write(System.lineSeparator());
+            } else if (userInputs.get(i).getCode().equals("D")) {
+                String lineToAppend = userInputs.get(i).getCode() + DELIMITER + userInputs.get(i).getDoneValue() +
+                        DELIMITER + userInputs.get(i).getDescription() + DELIMITER + userInputs.get(i).getBy();
+                fw.write(lineToAppend);
+                fw.write(System.lineSeparator());
+            } else if (userInputs.get(i).getCode().equals("E")) {
+                String lineToAppend = userInputs.get(i).getCode() + DELIMITER + userInputs.get(i).getDoneValue() +
+                        DELIMITER + userInputs.get(i).getDescription() + DELIMITER + userInputs.get(i).getAt();
+                fw.write(lineToAppend);
+                fw.write(System.lineSeparator());
+            }
+        }
+        fw.close();
+
+        System.out.println(SEPARATOR + "\nBye. I'm going back to sleep... zzz");
+    }
+
+    /**
+     * List out the tasks in the array userInputs with numbering
+     * @param line the user input
+     * @throws DukeException command list is invalid
+     */
+    public static void listTasks(String line) throws DukeException {
+        if (!line.equals("list")) {
+            throw new DukeException("command list does not take additional parameters");
+        }
+
+        System.out.println(SEPARATOR + "\n\tTasks to do... so lazy:");
+
+        for (int i = 1; i <= userInputsCount; i++) {
+            System.out.println("\t" + i + "." + userInputs.get(i - 1));
+        }
+        System.out.println(SEPARATOR);
+    }
+
+    /**
+     * Marks a particular task in userInputs as done
+     * @param line the user input
+     * @throws DukeException command done has invalid parameters
+     */
+    public static void markDone(String line) throws DukeException {
+        String[] taskInputs = line.split(" ");
+
+        if (taskInputs.length != 2) {
+            throw new DukeException("incorrect number of parameters for command done");
+        }
+
+        final int INDEX_DONE = 1;
+        int taskIndexDone = Integer.parseInt(taskInputs[INDEX_DONE]) - 1;
+
+        userInputs.get(taskIndexDone).markAsDone();
+
+        System.out.println(
+                SEPARATOR + "\n\tfinished this task... I need a break:\n\t\t" +
+                userInputs.get(taskIndexDone));
```

Comment:
There may be an extra newline here.
```suggestion
        System.out.println(SEPARATOR + "\n\tfinished this task... I need a break:\n\t\t" +
                userInputs.get(taskIndexDone));
```